### ISSUE TITLE
_beforeTokenTransfer hook for ERC721 is different and missing logic vs. non-upgradeable contract

### ISSUE THREAD

**Issue by thedavidmeister** [OpenZeppelin/openzeppelin-contracts-upgradeable#178]
normal hook:

```solidity
    function _beforeTokenTransfer(
        address from,
        address to,
        uint256, /* firstTokenId */
        uint256 batchSize
    ) internal virtual {
        if (batchSize > 1) {
            if (from != address(0)) {
                _balances[from] -= batchSize;
            }
            if (to != address(0)) {
                _balances[to] += batchSize;
            }
        }
    }
```

upgradeable hook:

```solidity
    function _beforeTokenTransfer(
        address from,
        address to,
        uint256 tokenId
    ) internal virtual {}
```

is this intentional?

**Comment by Amxx** [comment]
Hello @thedavidmeister 

The upgradeable contracts are automatically transpiled from the non-upgradeable ones. There is probably an issue with transpilation that caused the latest changes to not be reflected here. I'll look into that !

**Comment by Amxx** [comment]
This was fixed by #a78956e979e126e942d8bdf8a6ed693f57c19c0e
